### ISSUE TITLE
API: child oauth scopes are not authorizable unless explicitly selected during app registration

### ISSUE THREAD

**Issue by trwnh** [mastodon/mastodon#22140]
### Steps to reproduce the problem

1. register an app with e.g. `read write follow push`
2. try to get an authorization code for `write:statuses`

### Expected behaviour

it works, because the app was registered with `write`

### Actual behaviour

it fails, with invalid scope

### Detailed description

this is tricky because i assume they are on some level different scopes and doorkeeper is behaving as intended, but it creates a situation where apps have to register with all fine-grained child scopes that they ever intend to authorize for

### Specifications

mastodon.social

**Comment by ThisIsMissEm** [comment]
I don't believe this is a bug, perhaps an enhancement, but I'd be inclined not to fix. 

If you want the `write:statuses` scope, then you need to register the app with the `write:statuses` scope — the fact that we have these top level scopes that actually apply more broadly is something of an anti-pattern. We should (and do) always encourage using the most specific scopes possible for your OAuth application.
